### ISSUE TITLE
For initializations of indexed properties, IPropertyInitializer does not expose the arguments specified for the index.

### ISSUE THREAD

**Issue by jinujoseph** [dotnet/roslyn#18709]
Reported by @evildour  Refer https://github.com/dotnet/roslyn-analyzers/pull/1142#issuecomment-294216781

**Comment by evildour** [comment]
Here is some code which reproduces the issue:
```csharp
var tree = CSharpSyntaxTree.ParseText(@"
using System.Collections.Generic;
class C
{
    void Foo()
    {
        var x = new Dictionary<int, int> { [1] = 2 };
    }
}
",
CSharpParseOptions.Default.WithFeatures(
    new[] { new KeyValuePair<string, string>("IOperation", "true") }));
var compilation = CSharpCompilation.Create(
    "out.dll",
    new[] { tree },
    new[] { MetadataReference.CreateFromFile(typeof(int).Assembly.Location) },
    new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
var semanticModel = compilation.GetSemanticModel(tree);
var objectCreationExpression =
    tree.GetRoot().DescendantNodes().OfType<ObjectCreationExpressionSyntax>().First();
var operation = (IObjectCreationExpression)semanticModel.GetOperation(objectCreationExpression);
foreach (var initializer in operation.MemberInitializers)
{
    // There is no way to get the index arguments used here.
}
```

**Comment by mavasani** [comment]
Fixed with #20689
